### PR TITLE
feat: data table accessibility fixes

### PR DESCRIPTION
**Pull request by tulasigudibanda**
## Summary

Brief summary describing the changes here.

## ADO Story or GitHub Issue Link

https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/1942429

## Notes

Added roles for the table elements
Made freeze button accessible to screen reader
Added assistive text for sorting columns
Added assistive text for expanding/collapsing rows

## Checklist

- [ ] Used Conventional Commit messages as outlined in the contributing guide.
  - [ ] Noted breaking changes (if any).
- [ ] Documented/updated all props, events, slots, parts, etc with JSDoc.
  - [ ] Ran the `analyze` command to update Storybook docs.
- [ ] Added/updated Stories with controls to cover all variants.
- [ ] Ran `test` locally to address any failures.
- [ ] Added/updated the Figma link for the Story's Design tab.
- [ ] Added any new component exports to the src/index.ts file

## Testing Instructions

For checkbox selection/deselection, spacebar is standard. Checkboxes are native <input> , so there's no custom code, they work as the browser intends.

### CODE REVIEW

File: src/components/reusable/table/table-row.ts [hunk 1]
```diff
@@ -197,6 +218,7 @@ export class TableRow extends LitElement {
       )
     ) {
       this.expanded = expanded;
+      this.assistiveText = `${expanded ? 'Expanded' : 'Collapsed'}`;
```
Comment (brian-patrick-3) [inline src/components/reusable/table/table-row.ts:221]: We can't hard-code any strings, they must be customizable for localization purposes.

File: src/components/reusable/table/table-row.ts [hunk 2]
```diff
@@ -28,6 +28,18 @@ import { TableCell } from './table-cell';
...
+   * @internal
+   */
+  @property({ type: String, reflect: true })
+  override role = 'row';
+
+  /** aria-disabled.
+   * @internal
+   */
+  @property({ type: String, reflect: true })
+  'aria-disabled' = 'true';
```
Comment (brian-patrick-3) [inline src/components/reusable/table/table-row.ts:41]: Maybe should be `'false'` by default.

File: src/components/reusable/table/table.stories.ts [hunk 3]
```diff
@@ -223,7 +223,9 @@ export const ExpandableRows: Story = {
               <kyn-expanded-tr .colSpan=${8}>
                 <div class="center-content">
                   <!-- Put your expanded table content here -->
-                  <h4>Expanded content goes here</h4>
+                  <kd-button>
+                    <h4>Expanded content goes here</h4>
+                  </kd-button>
```
Comment (brian-patrick-3) [inline src/components/reusable/table/table.stories.ts:228]: Any particular reason this example was turned into a button?

Comment (tulasigudibanda) [reply src/components/reusable/table/table.stories.ts]: Reverted

File: src/components/reusable/table/table-row.ts (hunk 2 above)
Comment (tulasigudibanda) [reply src/components/reusable/table/table-row.ts]: Made false

File: src/components/reusable/table/table-row.ts (hunk 1 above)
Comment (tulasigudibanda) [reply src/components/reusable/table/table-row.ts]: Added textStrings prop on kyn-tr

**Review by brian-patrick-3 — APPROVED** [review]

**Comment by srpriyankashetty** [comment]
While sorting, when navigated back and forth for the same column. The reader is missing out to read the current state the column.
Discussed with Tulasi

**Comment by tulasigudibanda** [comment]
> While sorting, when navigated back and forth for the same column. The reader is missing out to read the current state the column. Discussed with Tulasi

@srpriyankashetty , pushed the fix . Please check

**Review by srpriyankashetty — APPROVED** [review]
1. Verified that Data table title read by screen reader
2. Verified that screen reader - reads out the total number of rows and columns and 
3. reads out the row number, column number and column information.
4. Verified that the freeze column option function is available for keyboard-only users. 
5. Verified that user is able to click on Hidden Column and Visible Column Checkbox can be selected and deselected using the Enter button or Space Bar button.
6. Verified that already Freeze column status is read as frozen
7. Verified that sorting is read as sort by column name
8. Verified that reader reads out the columns are sorting in both ascending and descending order.

**Pull request merged**